### PR TITLE
feat: support `InterleaveExecNode` in the proto

### PR DESCRIPTION
**Pull request by liukun4515**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

**Comment by liukun4515** [comment]
@alamb @jackwener  PTAL

File: datafusion/proto/tests/cases/roundtrip_physical_plan.rs [hunk 1]
```diff
@@ -798,3 +800,34 @@ fn roundtrip_sym_hash_join() -> Result<()> {
     }
     Ok(())
 }
+
+#[test]
```
Comment (liukun4515) [inline datafusion/proto/tests/cases/roundtrip_physical_plan.rs:804]: fill the test for union and interleave

**Review by metesynnada — APPROVED** [review]
LGTM.

**Review by jackwener — APPROVED** [review]

**Comment by liukun4515** [comment]
Thanks @jackwener  @metesynnada

**Review by alamb — COMMENTED** [review]
Thank you @liukun4515

**Pull request merged**